teachopencadd/talktorials/T018_automated_cadd_pipeline/utils/helpers/dogsitescorer.py

```diff
@@ -0,0 +1,357 @@
+    - https://proteins.plus/help/dogsite_rest
+    """
+
+    class FileUpload:
+        URL = "https://proteins.plus/api/pdb_files_rest"
+        REQUEST_MSG = "pdb_file[pathvar]"
+        RESPONSE_MSG = {
+            "status": "status_code",
+            "status_codes": {"accepted": "accepted", "denied": "bad_request"},
+            "message": "message",
+            "url_of_id": "location",
+        }
+        RESPONSE_MSG_FETCH_ID = {"message": "message", "id": "id"}
+
+    class SubmitJob:
+        URL = "https://proteins.plus/api/dogsite_rest"
+        QUERY_HEADERS = {
+            "Content-type": "application/json",
+            "Accept": "application/json",
+        }
+
+        RESPONSE_MSG = {"url_of_job": "location"}
+
+        RESPONSE_MSG_FETCH_BINDING_SITES = {
+            "result_table": "result_table",
+            "pockets_pdb_files": "residues",
+            "pockets_ccp4_files": "pockets",
+        }
+
+
+def upload_pdb_file(filepath):
+    """
+    Upload a PDB file to the DoGSiteScorer webserver using their API
+    and get back a dummy PDB code, which can be used to submit a detection job.
+
+    Parameters
+    ----------
+    filepath : str or pathlib.Path
+        Relative or absolute path of the PDB file.
+
+    Returns
+    -------
+    str
+        Dummy PDB code of the uploaded structure, which can be used instead of a PDB code.
+    """
+
+    filepath = Path(filepath)
+
+    # Read API URL and API request message from Constants
+    url = APIConsts.FileUpload.URL
+    request_msg = APIConsts.FileUpload.REQUEST_MSG
+
+    # Open the local PDB file for reading in binary mode
+    with open(filepath, "rb") as f:
+        # Post API query and get the response
+        response = requests.post(url, files={request_msg: f})
+    # Raise HTTPError if one occured during query
+    response.raise_for_status()
+    # Turn the response values into a dict
+    response_values = response.json()
+    if response.ok:
+        # If the request is accepted, the response will contain a URL,
+        # from which the needed ID of the uploaded protein can be obtained.
+        # Here, we store this URL from the response values in the url_of_id variable.
+        url_of_id = response_values[APIConsts.FileUpload.RESPONSE_MSG["url_of_id"]]
+    else:
+        raise ValueError(
+            "Uploading PDB file failed.\n"
+            + f"The response values are as follows: {response_values}"
+        )
+    # After getting the URL, it may take some time for the server to process the uploaded file
+    # and return an ID. Thus, we try 30 times in intervals of 5 seconds to query the URL,
+    # until we get the ID
+    for try_nr in range(30):
+        # Query the URL containing the ID
+        id_response = requests.get(url_of_id)
+        # Turn the response values into a dict
+        id_response_values = id_response.json()
+        # The response should contain the `id` keyword:
+        if id_response.ok & (
```

Comment:
`&` is not `and`. Are you sure you meant bitwise comparison?

Reply:
In this case, it doesn't matter whether we use `and` or `&`, since both conditions are boolean expressions. But for consistency I changed it to `and` anyway (see 155f328).